src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java

```diff
@@ -72,69 +74,68 @@ public String searchProductDescription(@RequestParam String description, Model m
-        return getProductsSortedByPrice(model, productList);
+        disableNameSorting();
+        reversePriceSorting();
+        searchService.setArchivedView(true);
+        List<Product> productList = productService.findAllProductsBySearchModel().getProducts();
+        return setTemplateModel(model, productList);
     }
 
     private List<Product> applySearch(boolean isSearchResultView) {
         searchService.setSearchResultView(isSearchResultView);
         searchService.setArchivedView(false);
-        List<Product> productList = productListCacheService.findBySearchRequest(searchService.getSearchModel()).getProducts();
-        List<Product> filteredList = productListCacheService.filterByPrice(productList, searchService.getSearchModel()).getProducts();
-        return hasDescription()
-                ? productListCacheService.searchForProductDescription(filteredList, searchService.getSearchModel()).getProducts()
-                : filteredList;
+        return productService.findAllProductsBySearchModel().getProducts();
     }
 
-    private boolean hasDescription() {
-        return null != searchService.getSearchModel().getDescription();
-    }
-
-    private String getProductsSortedByName(Model model, List<Product> productList) {
-        List<Product> sortedProducts = productListCacheService.sortListByName(productList, searchService.getSearchModel()).getProducts();
-        reverseNameSort();
-        return setTemplateModel(model, sortedProducts);
-    }
-
-    private String getProductsSortedByPrice(Model model, List<Product> productList) {
-        List<Product> sortedProducts = productListCacheService.sortListByPrice(productList, searchService.getSearchModel()).getProducts();
-        reversePriceSort();
-        return setTemplateModel(model, sortedProducts);
-    }
-
-    private void reverseNameSort() {
+    private void reverseNameSorting() {
         boolean sortAscending = searchService.getSearchModel().isSortAscendingName();
         searchService.getSearchModel().setSortAscendingName(!sortAscending);
     }
 
-    private void reversePriceSort() {
+    private void reversePriceSorting() {
         boolean sortAscending = searchService.getSearchModel().isSortAscendingPrice();
         searchService.getSearchModel().setSortAscendingPrice(!sortAscending);
     }
 
     private String setTemplateModel(Model model, List<Product> productList) {
+        ProductListDto productListDto = new ProductListDto(productList);
+        ProductDtoList productDtoListDto = productMapperUtil.convertToProductDtoList(productListDto);
         model.addAttribute(SEARCH_MODEL, searchService.getSearchModel());
-        model.addAttribute(PRODUCTS, productList);
+        model.addAttribute(PRODUCTS, productDtoListDto);
         model.addAttribute("cart", cartService.getCart());
         return PRODUCT_OVERVIEW;
     }
 
+    private void disableNameSorting() {
```

Comment:
both methods can be removed. This breaks the sorting-mechanism that was in place. Utilize the reverse****Sorting methods instead.

Reply:
removed